main/boost/APKBUILD

```diff
@@ -71,14 +71,12 @@ _libs="
 	wserialization
 	"
 for _lib in $_libs; do
-	_skip_lib=0
-	for _option in $_options_carch; do
-		case "$_option" in --without-${_lib}) _skip_lib=1; break ;; esac
-	done; unset -v _option
-	if [ ${_skip_lib:-0} -eq 0 ]; then
-		subpackages="$subpackages ${pkgname}-${_lib}:_boostlib"
-	fi
-done; unset -v _lib _skip_lib
+	# the trailing space in the below case is intentional so that we can match the last word
+	case "${_options_carch//$'\n'/ } " in
```

Comment:
I have doubts about `"${_options_carch//$'\n'/ } "`. It looks like it solves a problem in case `_options_carch` has new lines. It currently does not have newlines, and it will hopefully never need it either, since we want keep build options same across architectures.

It also adds bashism. Even if we tolerate this kind of variable substitution, I want to keep bashisms to a minimal, in case we decide to be strictly POSIX shell.

And finally, does it really work in busybox ash? I will have to verify that it actually works before I can merge it.

So my thinking here is: keep it simple. If we ever will need deal with newlines in `_options_carch` then lets deal with it at that time.

Reply:
The for loop version is more flexible in handling spaces, tabs and newline characters. You could use a function to print a newline instead and/or call out to see/awk/tr to format the case argument instead.

You are very welcome to double-check that BusyBox sh works with this case statement.

I would very much like to avoid breaking in unexpected ways when the value of `_oprions_carch` is split across lines.